cpp/tests/datetime/datetime_ops_test.cpp

```diff
@@ -827,4 +843,76 @@ TEST_F(BasicDatetimeOpsTest, TestQuarter)
   CUDF_TEST_EXPECT_COLUMNS_EQUAL(*extract_quarter(timestamps_s), quarter);
 }
 
+TYPED_TEST(TypedDatetimeOpsTest, TestFloorDatetime)
+{
+  using T = TypeParam;
+  using namespace cudf::test;
+  using namespace cudf::datetime;
+  using namespace cuda::std::chrono;
+
+  auto start = milliseconds(-2500000000000);  // Sat, 11 Oct 1890 19:33:20 GMT
+  auto stop_ = milliseconds(2500000000000);   // Mon, 22 Mar 2049 04:26:40 GMT
+
+  auto input = generate_timestamps<T>(this->size(), time_point_ms(start), time_point_ms(stop_));
+
+  auto host_val                     = to_host<T>(input);
+  thrust::host_vector<T> timestamps = host_val.first;
+
+  thrust::host_vector<T> floored_day(timestamps.size());
+  thrust::transform(timestamps.begin(), timestamps.end(), floored_day.begin(), [](auto i) {
+    return time_point_cast<typename T::duration>(floor<days>(i));
+  });
+  auto expected_day = fixed_width_column_wrapper<T, typename T::duration::rep>(floored_day.begin(),
+                                                                               floored_day.end());
+  CUDF_TEST_EXPECT_COLUMNS_EQUAL(*floor_day(input), expected_day);
+
+  thrust::host_vector<T> floored_hour(timestamps.size());
+  thrust::transform(timestamps.begin(), timestamps.end(), floored_hour.begin(), [](auto i) {
```

Comment:
Any particular reason that this test and only this test uses thrust while the rest use std?
```suggestion
  std::vector<T> floored_hour(timestamps.size());
  std::transform(timestamps.begin(), timestamps.end(), floored_hour.begin(), [](auto i) {
```

Reply:
Also applies to `floored_day` above - not just this one for hours. Good suggestion.